src/lib/utilities/click-outside.ts

```diff
@@ -0,0 +1,15 @@
+export function clickOutside(node: any) {
+  const handleClick = (event: MouseEvent) => {
+    if (node && !node.contains(event.target) && !event.defaultPrevented) {
+      node.dispatchEvent(new CustomEvent('click_outside', node));
```

Comment:
I wonder if we need a custom event? A click event should open it. A click event on the `document` and if the target is *not* the dropdown, then we should close it.

Reply:
will remake it to use without the custom event